spot fulfillment build failing with bqds-shared library

**Describe the bug**
The spot fulfillment docker and cloud build is failing with the *bqds-shared* library. Spot fulfillment has the local dependency and it is not available in the current build context

**To Reproduce**
Steps to reproduce the behavior:
1. Go to [api](../tree/master/api#deploy-cloud-run)
2. Run through the Cloud Build instructions
4. See error 

```
RUN npm install --only=production:
npm ERR! code ENOLOCAL
npm ERR! Could not install from "../shared" as it does not contain a package.json file.
```

**Expected behavior**
Builds successfully and no errors

**Screenshots**
n/a

**Desktop (please complete the following information):**
 - OS: Darwin 18.7.0

**Additional context**
Reported by @ramfordt

--- a/api/README.md
+++ b/api/README.md
@@ -290,16 +290,19 @@ Deploy with Cloud Run allows stateless HTTP containers on a fully managed enviro
 _Cloud Run and Cloud Build APIs will need to be enabled in your GCP project._
 
 Build with Cloud Build and TAG:
+_Cloud Build needs to run from parent directory for build context_
 
-    gcloud builds submit --tag gcr.io/${PROJECT_ID}/bqds-spot-fulfillment-api:${TAG}
+    cd ../../
+    gcloud builds submit --config api/v1alpha/cloudbuild.yaml --substitutions=TAG_NAME=${TAG}
 
 Deploy with Cloud Run Beta:
-_Note_ - There are a few environment variables that need to be set before the application starts (see below). [gcloud run deploy](https://cloud.google.com/sdk/gcloud/reference/beta/run/deploy#--set-env-vars) provides details for how they are set.
+_Note_ - There are a few environment variables that need to be set before the application starts (see below). [gcloud run deploy](https://cloud.google.com/sdk/gcloud/reference/run/deploy#--set-env-vars) provides details for how they are set.
 
-    gcloud beta run deploy bqds-spot-fulfillment-api \
+    gcloud run deploy bqds-spot-fulfillment-api \
       --image gcr.io/${PROJECT_ID}/bqds-spot-fulfillment-api:${TAG} \
       --region=us-central1 \
       --allow-unauthenticated \
+      --platform managed \
       --service-account ${SERVICE_ACCOUNT_NAME}@${PROJECT_ID}.iam.gserviceaccount.com \
       --set-env-vars=FULFILLMENT_CONFIG_BUCKET_NAME=${BUCKET_NAME} \
       --set-env-vars=FULFILLMENT_CONFIG_DESTINATION_PROJECT_ID=${PROJECT_ID} \
@@ -309,7 +312,7 @@ _Note_ - There are a few environment variables that need to be set before the ap
 
 Open the app URL in your browser. You can return the FQDN via:
 
-    gcloud beta run services describe bqds-spot-fulfillment-api --format="value(status.domain)"
+    gcloud run services describe bqds-spot-fulfillment-api --platform managed --format="value(status.domain)"
 
 ### Deploy App Engine
 

--- a/api/v1alpha/Dockerfile
+++ b/api/v1alpha/Dockerfile
@@ -25,11 +25,18 @@ MAINTAINER Chris Page <chrispage@google.com>
 RUN mkdir -p /app
 WORKDIR /app
 
+# Create the shared depdendency directory
+RUN mkdir -p /shared
+
 ## ---- Dependencies ----
 FROM base as dependencies
 # A wildcard is used to ensure both package.json AND package-lock.json are copied
 # where available (npm@5+)
-COPY package*.json ./
+# install shared dependencies first
+COPY shared/package*.json /shared/
+RUN cd /shared && npm install && cd -
+
+COPY api/v1alpha/package*.json ./
 # install app dependencies including 'devDependencies'
 RUN npm install
 
@@ -37,14 +44,21 @@ RUN npm install
 FROM dependencies AS build
 WORKDIR /app
 
+# Bundle shared dep
+COPY shared/*.js /shared/
 # Bundle app source
-COPY src/*.js /app/
+COPY api/v1alpha/src/*.js /app/
 
 # --- Release with Alpine ----
 FROM node:12.6-alpine as release
 WORKDIR /app
-COPY --from=dependencies /app/package.json ./
+# Install the shared dependencies first
+COPY --from=dependencies /shared/package.json /shared/
+RUN cd /shared && npm install --only=production && cd -
+COPY --from=build /shared /shared
+
 # Install app dependencies
+COPY --from=dependencies /app/package.json ./
 RUN npm install --only=production
 COPY --from=build /app ./
 EXPOSE 5555

--- a/api/v1alpha/cloudbuild.yaml
+++ b/api/v1alpha/cloudbuild.yaml
@@ -1,0 +1,10 @@
+steps:
+- name: "gcr.io/cloud-builders/docker"
+  args:
+  - build
+  - "--tag=gcr.io/$PROJECT_ID/bqds-spot-fulfillment-api:$TAG_NAME"
+  - "--file=./api/v1alpha/Dockerfile"
+  - .
+images:
+  - "gcr.io/$PROJECT_ID/bqds-spot-fulfillment-api:$TAG_NAME"
+

--- a/api/v1alpha/skaffold.yaml
+++ b/api/v1alpha/skaffold.yaml
@@ -18,7 +18,9 @@ kind: Config
 build:
   artifacts:
   - image: bqds-spot-fulfillment-api
-    context: .
+    context: ../../.
+    docker:
+      dockerfile: api/v1alpha/Dockerfile
 deploy:
   kubectl:
     manifests:

